src/crc_prediction_markets/LiquidityVaultToken.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {ERC1155 as OZERC1155} from "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract LiquidityVaultToken is OZERC1155, AccessControl {
```

Comment:
Contract for managing/holding LP tokens for `LiquidityAdder` and `LiquidityRemover` - note that fpmm.addFunding and fpmm.removeFunding requires 
```
balances[msg.sender] += shares
```
hence we need a separate entity holding the LP tokens. This also explains the transfer between vault <> adder and vault <> remover.